### PR TITLE
Adds ControllerBase to Mvc.Core

### PR DESCRIPTION
**Pull request by rynowak**
This change adds a base class for controllers to Mvc.Core that can be used
without a dependency on views, json, etc.

Most of the functionality on Controller moves to ControllerBase. I kept
the IActionFilter and IDisposable functionality on Controller since it's
not really a fit with the 'minimal' philosophy.

### CODE REVIEW

**Comment by rynowak** [comment]
Fixes #3094

File: src/Microsoft.AspNet.Mvc.Core/ControllerBase.cs [hunk 1]
```diff
@@ -0,0 +1,1451 @@
...
+using Microsoft.AspNet.Mvc.ModelBinding.Validation;
+using Microsoft.AspNet.Mvc.Routing;
+using Microsoft.AspNet.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Base class for an MVC controller.
```
Comment (Eilon) [inline src/Microsoft.AspNet.Mvc.Core/ControllerBase.cs]: Need to have a different description from `Controller`. In fact, should update both of them to say something unique about what each one is (and perhaps isn't).

**Comment by Eilon** [comment]
For some reason GitHub isn't showing the whole diff, but I assume that 99.9% of this is cut & paste from Controller to ControllerBase.

**Comment by rynowak** [comment]
> 23 additions, 1,458 deletions not shown

That's what github has to say about the changes to `Controller.cs`. It's all just cut-paste

**Comment by Eilon** [comment]
Sure :shipit: after fixing comment. But I'd like to see the updated comments first :smile:

**Comment by rynowak** [comment]
Updated

**Pull request closed without merging**